modules/src/core/ics04_channel/packet.rs

```diff
@@ -105,7 +123,7 @@ pub struct Packet {
     pub destination_port: PortId,
     pub destination_channel: ChannelId,
     #[serde(serialize_with = "crate::serializers::ser_hex_upper")]
-    pub data: Vec<u8>,
+    pub data: PacketData,
```

Comment:
Could the `Debug` impl of `Packet` be hand-implemented (with all the help you can get from `DebugStruct` and a temporary wrapper struct) rather than introducing a newtype for the member?

Reply:
Good idea, I didn't know about `DebugStruct`. The only downside that I see to that approach would be that when we modify `Packet`, we need to remember to go add that field to the debug output. I still think it's a cleaner approach, thanks for the suggestion!

It's probably easier to close this PR and open a different one, instead of reverting all the changes I made.